File: src/test/java/de/pflugradts/passbird/application/security/CryptoProviderFaker.java
package de.pflugradts.passbird.application.security;

import de.pflugradts.passbird.domain.model.transfer.Bytes;
import de.pflugradts.passbird.domain.service.password.encryption.CryptoProvider;
import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.lenient;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class CryptoProviderFaker {

    private CryptoProvider cryptoProvider;

    public static CryptoProviderFaker faker() {
        return new CryptoProviderFaker();
    }

    public CryptoProviderFaker forInstance(final CryptoProvider cryptoProvider) {
        this.cryptoProvider = cryptoProvider;
        return this;
    }

    public CryptoProviderFaker withMockedEncryption() {
        lenient().when(cryptoProvider.encrypt(any(Bytes.class))).thenAnswer(
                invocation -> invocation.getArgument(0));
        lenient().when(cryptoProvider.decrypt(any(Bytes.class))).thenAnswer(
                invocation -> invocation.getArgument(0));
        return this;
    }

    public CryptoProvider fake() {
        return cryptoProvider;
    }

}

File: src/test/kotlin/de/pflugradts/passbird/application/security/CryptoProviderFaker.kt
package de.pflugradts.passbird.application.security

import de.pflugradts.passbird.domain.model.shell.EncryptedShell
import de.pflugradts.passbird.domain.model.shell.Shell
import de.pflugradts.passbird.domain.model.shell.fakeDec
import de.pflugradts.passbird.domain.model.shell.fakeEnc
import de.pflugradts.passbird.domain.service.password.encryption.CryptoProvider
import io.mockk.every

fun fakeCryptoProvider(instance: CryptoProvider) {
    every { instance.encrypt(any<Shell>()) } answers { firstArg<Shell>().fakeEnc() }
    every { instance.decrypt(any<EncryptedShell>()) } answers { firstArg<EncryptedShell>().fakeDec() }
}
